packages/utils/src/document/reconversion-kickoff-loader.ts

```diff
@@ -0,0 +1,159 @@
+): Promise<void> {
+  const parser = new JSONParser({ stringBufferSize: undefined, paths: ["$.*"] });
+  parser.onValue = onValue;
+  await new Promise((resolve, reject) => {
+    const inputStream = fs.createReadStream(path, { encoding: "utf8" });
+    inputStream.on("error", reject);
+    parser.onError = reject;
+    parser.onEnd = () => resolve(undefined);
+    inputStream.on("data", chunk => parser.write(chunk));
+    inputStream.on("end", () => parser.end());
+  });
+}
+
+async function main() {
+  const patients: Array<{ patient_id: string; cx_id: string }> = [];
+
+  console.log("Loading patient data from file...");
+  await loadDataFromLargeJsonFile(fileName, ({ value }) => {
+    const patient = value as { patient_id: string; cx_id: string };
+    patients.push(patient);
+  });
+
+  console.log(`Loaded ${patients.length} patients from file`);
+
+  const patientsByCxId = groupBy(patients, "cx_id");
+
+  const payloads: ReconversionKickoffParams[] = [];
+
+  for (const [cxId, cxPatients] of Object.entries(patientsByCxId)) {
+    console.log("cxId", cxId);
+
+    const patientIds = Array.from(new Set(cxPatients.map(p => p.patient_id)));
+
+    const patientChunks = chunk(patientIds, PATIENT_CHUNK_SIZE);
+    for (const [i, patients] of patientChunks.entries()) {
+      console.log(`Chunk ${i + 1} of ${patients.length}`);
+      const payloadParams: ReconversionKickoffParams = {
+        messageId: uuidv7(),
+        cxId,
+        patientIds: patients,
+        dateFrom,
+        ...(dateTo && dateTo !== "" ? { dateTo } : {}),
+      };
+      payloads.push(payloadParams);
+    }
+  }
+
+  console.log(`Total payloads to send: ${payloads.length}`);
+
+  // Chunk payloads into batches to respect SQS rate limits
+  const payloadBatches = chunk(payloads, SQS_BATCH_SIZE);
+  console.log(`Sending in ${payloadBatches.length} batches of max ${SQS_BATCH_SIZE} messages each`);
+
+  let totalSent = 0;
+  let totalErrors = 0;
+
+  for (const [batchIndex, batch] of payloadBatches.entries()) {
+    console.log(
+      `Processing batch ${batchIndex + 1}/${payloadBatches.length} (${batch.length} messages)`
+    );
+
+    const batchPromises = batch.map(async p => {
+      const payloadString = JSON.stringify(p);
+      try {
+        // For debugging - write to file
+        // fs.appendFile("sqsMessages.json", payloadString + "\n", "utf8", () => {});
+
+        await sqsClient.sendMessageToQueue(sqsUrl, payloadString, {
+          fifo: true,
+          messageDeduplicationId: createUuidFromText(payloadString),
+          messageGroupId: p.cxId,
+        });
+
+        return { success: true };
+      } catch (e) {
+        console.error(`Error sending message: ${e}`);
+        return { success: false, error: e };
+      }
+    });
+
+    const batchResults = await Promise.all(batchPromises);
+
+    const batchSuccesses = batchResults.filter(r => r.success).length;
+    const batchErrors = batchResults.filter(r => !r.success).length;
```

Comment:
This is only needed for Promise.allSettled right? IMO no need to add this here, I'd just call Promise.all